flocker/control/_model.py

```diff
@@ -110,23 +111,33 @@ def pmap_field(
     :param initial: An initial value for the field.  This will first be coerced
         using the field's factory.  If not given, the initial value is an empty
         map.
+    :param factory: A factory used to convert input arguments to the stored
+        value whenever it is set. Note that this will be composed with the
+        constructor for the ``CheckedPMap`` class constructed for this field.
```

Comment:
Should this work?:

```
class Foo(PClass):
    bar = pmap_field(key_type=bytes, value_type=int, factory=int)


In [58]: Foo().bar.set('moo', .1)
---------------------------------------------------------------------------
CheckedValueTypeError                     Traceback (most recent call last)
<ipython-input-58-5a65a2a9ff99> in <module>()
----> 1 Foo().bar.set('moo', .1)

/home/richard/.virtualenvs/4417/lib/python2.7/site-packages/pyrsistent/_pmap.pyc in set(self, key, val)
    157         pmap({'a': 1, 'c': 4, 'b': 2})
    158         """
--> 159         return self.evolver().set(key, val).persistent()
    160 
    161     def remove(self, key):

/home/richard/.virtualenvs/4417/lib/python2.7/site-packages/pyrsistent/_checked_types.pyc in set(self, key, value)
    458         def set(self, key, value):
    459             _check_types([key], self._destination_class._checked_key_types, self._destination_class, CheckedKeyTypeError)
--> 460             _check_types([value], self._destination_class._checked_value_types, self._destination_class)
    461             self._invariant_errors.extend(data for valid, data in (invariant(key, value)
    462                                                                    for invariant in self._destination_class._checked_invariants)

/home/richard/.virtualenvs/4417/lib/python2.7/site-packages/pyrsistent/_checked_types.pyc in _check_types(it, expected_types, source_class, exception_type)
    171                     expected_types=tuple(get_type(et).__name__ for et in expected_types),
    172                     actual_type=actual_type.__name__)
--> 173                 raise exception_type(source_class, expected_types, actual_type, e, msg)
    174 
    175 

CheckedValueTypeError: Type StrIntPMap can only be used with ('int',), not float

In [59]: Foo().bar.set('moo', 1)
Out[59]: StrIntPMap({'moo': 1})

```

I expected the `int` factory to coerce the float to int before it gets validated and assigned.